File: examples/typescript/discovery.ts
import { BusinessTravelAPIApi, Configuration } from "@myescapeplan/sdk";

const apiKey = process.env.MYESCAPEPLAN_API_KEY;
if (!apiKey) {
  throw new Error("Set MYESCAPEPLAN_API_KEY before running this example.");
}

const api = new BusinessTravelAPIApi(
  new Configuration({ apiKey }),
);

const discovery = await api.createDiscovery({
  idempotencyKey: `example-${Date.now()}`,
  discoveryRequest: {
    query: "four nights next month, somewhere warm, under 500 GBP",
    limit: 5,
  },
});

console.log({
  discoveryId: discovery.discoveryId,
  opportunities: discovery.opportunities,
});

File: examples/python/discovery.py
"""Minimal Travel Discovery API example."""

import asyncio
import os
import uuid

from myescapeplan import ApiClient, BusinessTravelAPIApi, Configuration
from myescapeplan.models.discovery_request import DiscoveryRequest


async def main() -> None:
    api_key = os.environ.get("MYESCAPEPLAN_API_KEY")
    if not api_key:
        raise RuntimeError("Set MYESCAPEPLAN_API_KEY before running this example.")

    configuration = Configuration()
    configuration.api_key["BusinessApiKey"] = api_key

    async with ApiClient(configuration) as client:
        api = BusinessTravelAPIApi(client)
        discovery = await api.create_discovery(
            idempotency_key=f"example-{uuid.uuid4()}",
            discovery_request=DiscoveryRequest(
                query="four nights next month, somewhere warm, under 500 GBP",
                limit=5,
            ),
        )

        print({
            "discovery_id": discovery.discovery_id,
            "opportunities": discovery.opportunities,
        })


if __name__ == "__main__":
    asyncio.run(main())
